### PR TITLE
Refactor/dialog

### PR DESCRIPTION
**Pull request by mandaiy**
there's much room to improve and i still need to fix some code
(e.g., there are hard coding strings)
please comment.

i know this is a huge pull req. please be patient.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -29,9 +29,11 @@ android {
     defaultConfig {
         applicationId "com.pileproject.drive"
         minSdkVersion 15
-        targetSdkVersion 22
+        targetSdkVersion 23
```
Comment (tiwanari) [inline app/build.gradle:32]: is this required? what for?

File: app/src/androidTest/java/com/pileproject/drive/util/fragment/AlertDialogFragmentTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    @Rule
+    public ActivityTestRule<TestActivity> mActivityRule = new ActivityTestRule<>(TestActivity.class);
+
+    @Before
+    public void setUp() throws Exception {
+        mActivity = mActivityRule.getActivity();
+    }
+
+    @Test
+    public void testCheckComponent_titleString() {
```
Comment (tiwanari) [inline app/src/androidTest/java/com/pileproject/drive/util/fragment/AlertDialogFragmentTest.java:52]: title?

File: app/build.gradle (hunk 1 above)
Comment (mandaiy) [inline app/build.gradle:32]: No strong opinion but some say it should be the latest
http://stackoverflow.com/questions/33947660/what-should-i-set-for-compilesdkversion-minsdkversion-and-targetsdkversion

**Comment by tiwanari** [comment]
OK. LGTM. APPROVED.

**Pull request merged**